spec/lib/two_factor_authentication/models/two_factor_authenticatable_spec.rb

```diff
@@ -90,6 +90,12 @@ def do_invoke(code, user)
         code = @totp_helper.totp_code(1.minutes.ago.to_i)
         expect(do_invoke(code, instance)).to eq(false)
       end
+
+      it 'prevents code reuse' do
+        code = @totp_helper.totp_code
+        expect(do_invoke(code, instance)).to eq(true)
+        expect(do_invoke(code, instance)).to eq(false)
+      end
```

Comment:
Could you also add a test that verifies that if you travel in time (with Timecop for example) the same amount of time as the configured drift (30 seconds by default), and then try to use the same code, it is invalid. This particular test failed when I used a different implementation of code reuse, so I want to make sure this solution works in this scenario as well.